src/arrayMethodSort.js

```diff
@@ -4,8 +4,31 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+    const compare =
+      compareFunction ||
+      function (a, b) {
+        if (String(a) > String(b)) {
+          return 1;
+        }
+
+        if (String(a) < String(b)) {
+          return -1;
+        }
+
+        return 0;
+      };
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - 1; j++) {
```

Comment:
The inner loop doesn't need to iterate over the entire array. Each pass of the outer loop 'bubbles' the largest element to the end of the array, so the inner loop can stop one element earlier each time. This can slightly improve the performance of your function. Instead of `this.length - 1`, you can use `this.length - i - 1`.